src/otx/algo/common/utils/prior_generators/base_prior_generator.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Base prior generator."""
+
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import TYPE_CHECKING, Callable
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+
+class BasePriorGenerator:
```

Comment:
```suggestion
class BasePriorGenerator(ABC):
```
How about making it as abstract class explictly?

Reply:
Done c3dfef0